### DISCUSSION TITLE
Modification of Apache License and use of Open Source

### DISCUSSION THREAD

**Discussion by ssddanbrown** [caprover/caprover#2075] [General]
Hello,
I know some of these subjects have been raised before (#1852, #2035, #2036) but just want to add some extra information/context.

Before getting into it, I want to be clear I completely respect your right to license your efforts how you wish, I appreciate the amount of effort and the challenges of open source work, my main concern is about avoiding potential confusing or misleading language around open source and licensing.

First up, your license is essentially a modified Apache license. Apache are against this when keeping the license the name the same, as per [their license FAQ here](https://www.apache.org/foundation/license-faq.html#mod-license) since, when adding extra terms, you're essentially no longer distributing under the Apache 2.0 license as it's known and therefore that could be misleading or confusing to users.

Secondly, the project is [still referring to itself](https://github.com/caprover/caprover/blob/a563cd0dd593a7995b51e05fd64debed756677f6/TERMS_AND_CONDITIONS.md?plain=1#L5) as open source while under terms that restrict fundamental freedoms that are commonly expected in open source (and "free") software.
Again, this could be misleading if folks see this referred to as open source and then therefore expect it to be provided with the freedoms that the [commonly regarded open source definition](https://opensource.org/osd) provides. I've written further about why I think this is important [on my blog here](https://danb.me/blog/why-open-source-term-is-important/). "Source available" is a common alternative term that can fit the kind of terms provided here.

---

To specifically state how this does not fit into the OSD:

> Terms of Service / Terms & Conditions may change at any point without a prior notice.

This is awkward since users have to accept that before using as per your terms. Therefore rights of use, modification and distribution could be pulled out from under users at any time, and it also limits the application to a single point of authorship in a way.

> Unless provided with a written agreement, the paid features of CapRover (as a service) cannot be modified.

Modification is common expected right of open source (OSD criteria 3). From the code, it can be unclear what comes under this scope and there's a lot of room for interpretation here, making modification in respect to the license problematic.

> Unless provided with a written agreement, no persons are permitted to redistribute another paid version of CapRover.

Being able to charge & distribute is a common expected right of open source (OSD criteria 1).

> Furthermore, any modifications of free features of CapRover should be distributed as free & opensource software.

This rule is then in conflict with the above referenced statements and again put restrictions on distribution. You can't meet this when assuming "opensource" is as per its most common definition (OSD) since you already have to agree to (possibly changing terms) that are against the OSD. If not respecting the OSD, then its quite open to interpretation what that line actually means.

---

To add some more context into why I'm raising this, I've now come across two other projects that have used caprover as a reference, where the same confusing terms and referring as open source have been used:

- https://github.com/orgs/swiftwave-org/discussions/680
- https://github.com/Dokploy/dokploy/discussions/3

**Comment by githubsaturn** [comment]
Hi Dan, this was brought up in the past and the tag line was updated. See [here](https://github.com/caprover/caprover-website/commit/2bc96c44c67443b0051166918c9a30a2493c4baf)

The one in `Terms` slipped through the cracks and now updated: https://github.com/caprover/caprover/commit/63906469b855d9f71dae59edff2f2e1b03f15fc5
